### PR TITLE
Allow specifying another store via commandline option for the download command

### PR DESCRIPTION
**Pull request by morphis**
This allows specifying a store id for the snap download command to allow downloading snaps from branded stores. 

```
    $ snap download --store-id=xx my-custom-gadget
```

The implementation doesn't respect authentication yet so if the store isn't available for the public the download will still fail.

### CODE REVIEW

**Comment by morphis** [comment]
@mvo5 @fgimenez Do we have something like a test branded store I could use to extend the snap download spread test case for this?

File: cmd/snap/cmd_download.go [hunk 1]
```diff
@@ -114,7 +116,13 @@ func (x *cmdDownload) Execute(args []string) error {
 	var authContext auth.AuthContext
 	var user *auth.UserState
 
-	sto := store.New(nil, authContext)
+	var cfg *store.Config = nil
```
Comment (mvo5) [inline cmd/snap/cmd_download.go]: No need to init with `= nil` that is the default in go.

File: tests/main/snap-download/task.yaml [hunk 2]
```diff
@@ -25,6 +25,12 @@ execute: |
     su -l -c "SNAPPY_USE_STAGING_STORE=$SNAPPY_USE_STAGING_STORE snap download test-snapd-tools" test
     ls /home/test/test-snapd-tools_*.snap
     verify_asserts /home/test/test-snapd-tools_*.assert
+
+    echo "Snap download can download from a different store"
```
Comment (morphis) [inline tests/main/snap-download/task.yaml]: Thanks! Had the same pending locally but didn't know that there is a demo store we can use for testing.

**Comment by morphis** [comment]
@mvo5 thanks for the changes!

**Comment by pedronis** [comment]
I'm a bit vary of adding this as a command line option until we understand the auth story, I would be more comfortable with a env var  for now

**Comment by pedronis** [comment]
Also "snap download" already shares helpers with prepare-image,  there is probably more short term bang for bucks return by doing something at the level of image/ I have some ideas

**Comment by niemeyer** [comment]
Yes, this seems a bit like crossing some wires we shouldn't. The store so far has been something device-scoped, which manufacturers can get setup for themselves. Not something we want to switch back and forth at the CLI.

**Pull request closed without merging**